### PR TITLE
New App: Crowdfunding 🎉 🎉 

### PR DESCRIPTION
**Pull request by Kukks**
Quite excited with this one, it's been fun :D

With this PR, users will be able to create a crowdfunding campaign. 

A crowdfunding app allows you to create a landing page with a funding goal where users are able to contribute. Additionally, I've repurposed the existing template mechanism of the POS app so that you can allow users who contribute specific amounts to receive a perk. 

There are all sorts of configurable options making this super flexible and adjustable for multiple uses. Screenshots coming soon!

Feature list:
* Signalr real-time pushing
* No JS version
* Fun animations on payment received (optional)
* Fun sounds on payments received (optional)
* Rich text editor for description
* Contribution Perks
* Disqus Integration
* Softcap/ Hardcap options
* Multiple options on how contribution goal is calculated
* Reset goal every X hour/day/month/year
* Count all Inoivces in store to goal option

### CODE REVIEW

**Comment by NicolasDorier** [comment]
image

**Comment by NicolasDorier** [comment]
Can you try to break that a bit? Like one PR for adding your dependencies, one PR for making the refactoring about event's name, so that it scare me less? :D

**Comment by Kukks** [comment]
@NicolasDorier  https://github.com/btcpayserver/btcpayserver/pull/492

**Comment by NicolasDorier** [comment]
is this the last one to merge?

File: BTCPayServer/Views/Apps/UpdateCrowdfund.cshtml [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+                <a asp-action="ListInvoices" asp-controller="Invoice" asp-route-searchterm="@($"orderid:{CrowdfundHubStreamer.CrowdfundInvoiceOrderIdPrefix}{Model.AppId}")">Invoices generated by app</a>&nbsp
+                <a asp-action="ListApps">Back to the app list</a>
+            </div>
+        </div>
+    </div>
+</section>
+
+@section Scripts {
+    <link rel="stylesheet" href="~/vendor/highlightjs/default.min.css">
+    <script src="~/vendor/highlightjs/highlight.min.js"></script>
```
Comment (NicolasDorier) [inline BTCPayServer/Views/Apps/UpdateCrowdfund.cshtml]: You must use the bundling stuff here

File: BTCPayServer/Views/AppsPublic/Crowdfund/MinimalCrowdfund.cshtml [hunk 2]
```diff
@@ -0,0 +1,167 @@
+@using BTCPayServer.Models.AppViewModels
+@using Microsoft.CodeAnalysis.CSharp.Syntax
+@model  BTCPayServer.Models.AppViewModels.ViewCrowdfundViewModel
+
+@functions{
+
+    public bool Started()
```
Comment (NicolasDorier) [inline BTCPayServer/Views/AppsPublic/Crowdfund/MinimalCrowdfund.cshtml]: Can you just add this as part of the ViewCreowdFundViewModel?

File: BTCPayServer/Views/AppsPublic/Crowdfund/MinimalCrowdfund.cshtml [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+    {
+        return !Model.StartDate.HasValue || DateTime.Now.ToUniversalTime() > Model.StartDate;
+    }
+
+    public bool Ended()
```
Comment (NicolasDorier) [inline BTCPayServer/Views/AppsPublic/Crowdfund/MinimalCrowdfund.cshtml]: this too

File: BTCPayServer/wwwroot/crowdfund/services/audioplayer.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        var audioElem = new Audio(path);
+
+        audioElem.play().catch(function(){
+            debugger;
+        })
+    }
+}
+
+
+function playQuakeSound (name){
```
Comment (NicolasDorier) [inline BTCPayServer/wwwroot/crowdfund/services/audioplayer.js]: lol will have to make this customizable later :D

**Comment by NicolasDorier** [comment]
Well if we use sanitizer, why not using markdown instead?

**Comment by Kukks** [comment]
I mentioned markdown to @bitcoinshirt  before but I think we concluded that it was too geeky for the general public :P

**Comment by NicolasDorier** [comment]
true

File: BTCPayServer/Models/AppViewModels/UpdateCrowdfundViewModel.cs [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+        [Required] public string Description { get; set; }
+        
+        [Display(Name = "Featured Image")]
+        public string MainImageUrl { get; set; }
+
+        public string NotificationUrl { get; set; }
+
+        [Required]
+        [Display(Name = "Allow crowdfund to be publicly visible( still visible to you)")]
```
Comment (NicolasDorier) [inline BTCPayServer/Models/AppViewModels/UpdateCrowdfundViewModel.cs]: "Allow crowdfund to be publicly visible( still visible to you)" => "Allow crowdfund to be publicly visible (still visible to you)"

**Comment by NicolasDorier** [comment]
image

**Pull request merged**